service/src/main/kotlin/xyz/lebkuchenfm/domain/songs/SongsService.kt

```diff
@@ -63,4 +70,16 @@ class SongsService(
         )
         return songsRepository.insert(newSong).getOr(null)
     }
+
+    private suspend fun getOrCreateSongsCorrespondingToYoutubeVideos(youtubeVideos: List<YoutubeVideo>): List<Song> {
+        val youtubeIds = youtubeVideos.map { it.id }
+        val knownSongs = songsRepository.findByYoutubeIds(youtubeIds)
+        val knownIds = knownSongs.map { it.youtubeId }
+
+        val newSongs = youtubeVideos
+            .filterNot { it.id in knownIds }
+            .mapNotNull { createNewSong(it.id, it.name) }
+
+        return knownSongs + newSongs
```

Comment:
This way we're losing sorting from original playlist.

Reply:
very good catch!